@ChristophMajcenAtXxxlutz , the latest issue is:

```
Running mypy type checking ...02:12
src/confluent_kafka/schema_registry/_sync/avro.py:766: error:(B Argument 1 has incompatible type (B"str | list[Any] | dict[Any, Any] | None"(B; expected (B"str | list[Any] | dict[Any, Any]"(B  (B[arg-type](B02:17
src/confluent_kafka/schema_registry/_async/avro.py:775: error:(B Argument 1 has incompatible type (B"str | list[Any] | dict[Any, Any] | None"(B; expected (B"str | list[Any] | dict[Any, Any]"(B  (B[arg-type](B02:17
Found 2 errors in 2 files (checked 106 source files)(B
```

--- a/src/confluent_kafka/schema_registry/_async/avro.py
+++ b/src/confluent_kafka/schema_registry/_async/avro.py
@@ -739,12 +739,12 @@ class AsyncAvroDeserializer(AsyncBaseDeserializer):
         if isinstance(payload, bytes):
             payload = io.BytesIO(payload)
 
-        reader_schema: Optional[AvroSchema]
+        reader_schema: AvroSchema
         if latest_schema is not None and subject is not None:
             migrations = await self._get_migrations(subject, writer_schema_raw, latest_schema, None)
             reader_schema_raw = latest_schema.schema
             reader_schema = await self._get_parsed_schema(latest_schema.schema)
-        elif self._schema is not None:
+        elif self._schema is not None and self._reader_schema is not None:
             migrations = None
             reader_schema_raw = self._schema
             reader_schema = self._reader_schema

--- a/src/confluent_kafka/schema_registry/_sync/avro.py
+++ b/src/confluent_kafka/schema_registry/_sync/avro.py
@@ -730,12 +730,12 @@ class AvroDeserializer(BaseDeserializer):
         if isinstance(payload, bytes):
             payload = io.BytesIO(payload)
 
-        reader_schema: Optional[AvroSchema]
+        reader_schema: AvroSchema
         if latest_schema is not None and subject is not None:
             migrations = self._get_migrations(subject, writer_schema_raw, latest_schema, None)
             reader_schema_raw = latest_schema.schema
             reader_schema = self._get_parsed_schema(latest_schema.schema)
-        elif self._schema is not None:
+        elif self._schema is not None and self._reader_schema is not None:
             migrations = None
             reader_schema_raw = self._schema
             reader_schema = self._reader_schema

